### DISCUSSION TITLE
Filter option picker from previous option picker

### DISCUSSION THREAD

**Discussion by splerou** [Budibase/budibase#11028] [Help] [answered]
Hi,

We are trying to filter an option picker from a previous option picker in the same app/form. We aren't able to find a way to do so...

Example is when I select from first option filter the Environement : DEV/PROD

I want to filter the Patching schedule to : 

If DEV show next option picker => TUESDAY / WEDNESDAY
If PROD show next option picker => SATURDAY / SUNDAY 

Hope this is clear, we are news to this platform so sorry if this is super simple and we just haven't seen the option...

Thanks 
Splerou

**Comment by R2bEEaton** [comment, marked as answer]
Hello, yes! This should be possible using element conditions.

1. Set up your first options picker with an options source of Custom. Define the options to be either DEV or PROD.
2. Set up your second options picker. Choose the Custom options source, but **don't define any options**.
3. On this new options picker, click Conditions on the top right and Configure conditions.
4. Set up two conditions as shown. This will set custom options depending on whether the previous field is DEV or PROD.

![image](https://github.com/Budibase/budibase/assets/34921506/77e60b37-43ab-4298-a713-cd1f2bbfbe9d)

5. Make the field required so that users can't get away with leaving it blank in it (unless that's ok).

![image](https://github.com/Budibase/budibase/assets/34921506/b496bdde-26be-43a3-81e4-303528edc029)

6. And finally, go back to the first DEV / PROD field and add an On change action as shown. This will prevent users from picking PROD, then SUNDAY, then going back to DEV. Picking DEV + SUNDAY should be impossible after adding this action.

![image](https://github.com/Budibase/budibase/assets/34921506/951a7c9d-03d6-4016-8900-dcb905298751)

Here's an app export you can [import and explore](https://docs.budibase.com/docs/export-and-import-apps#import-app) if you need to see it in greater detail: 
[splerou-export-1688560798746.tar.gz](https://github.com/Budibase/budibase/files/11957924/splerou-export-1688560798746.tar.gz)

### Hope this helps!
Let me know if you have any further questions. Otherwise, make sure to mark this as the answer so that it can help future users! 😃

**Reply by melohagan** [reply]
This is a great write up Ryan - thank you! 😄

**Reply by splerou** [reply]
Hi @R2bEEaton,

First of all tahnks for you help, much apreciated. 

Unfortunetly it doesn't work on my end, the second option picker stays blank with no options to select :

![image](https://github.com/Budibase/budibase/assets/137413434/4f0eff8b-eaa5-4da3-a2a3-09c4f4de211f)

Any idea ? I followed the exact instructions on a blank app and on the exported app, I get the same result.

Splerou

**Reply by R2bEEaton** [reply]
Hi @splerou, hm. Could you send me an export of the blank app?

**Reply by splerou** [reply]
Hi,

[githubexportimport-export-1689603632051.tar.gz](https://github.com/Budibase/budibase/files/12070272/githubexportimport-export-1689603632051.tar.gz)

Here you go 

Thanks again

**Reply by R2bEEaton** [reply]
I imported this app and it seems to be working perfectly with no changes, so I'm not sure what could be causing the issue. Which version of Budibase are you on? I'm self-hosted on 2.7.14.

**Reply by splerou** [reply]
Hi,

I'm on self-hosted 2.7.21. I'll try installing others versions and let you know. 

Thanks

**Reply by splerou** [reply]
Hi,

I'm now on 2.8.10 latest version and same thing... maybe this was removed between 2.7.14 and 2.7.21...

Now I did the upgrade i'm going to look if new version has something similar.

Will let you know !

**Reply by R2bEEaton** [reply]
Hmm, I doubt it was removed. This has been possible for a while now. @melohagan I can't figure this one out, do you have any insight?

**Reply by melohagan** [reply]
Gave it a quick test on the latest and it looks like it works fine to me!

![Screenshot 2023-07-18 at 10 14 17](https://github.com/Budibase/budibase/assets/101575380/3a8bdb14-5fb1-4dcd-bd6e-beff91e7945f)

Using Google Chrome.

**Reply by splerou** [reply]
I've now tried all 3 versions... same result... I'm also using Chrome... Maybe soemthing wrong in my docker compose configuration, what do you think ?

**Reply by splerou** [reply]
Currently we have setup traefik as the reverse proxy in front of Budibase. Which looks like is working fine. But we recently had an issue with another app that wasn't showing the web page at all and traefik was the issue. We since moved to nginx proxy instead on that second app is now working fine. We are going to change the configuration and run Budibase with nginx instead and see. 

Will let you know

**Reply by splerou** [reply]
Hi,

It was the traefik reverse proxy that was causing the issue... we have now created it with the nginx reverse proxy instead and it works !!!

Thanks a lot for your help !! 
Splerou
